version_check.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+
+import sys
+from typing import Optional, Dict
+import tempfile
+
+import mistune
+import CppHeaderParser
+import semver
+import git
+
+
+def die(*args, error_code=1, **kwargs):
+    print('ERROR: ', end='', file=sys.stderr)
+    print(*args, **kwargs, file=sys.stderr)
+    exit(error_code)
+
+
+class SemVerWithVPrefix(semver.VersionInfo):
+    @classmethod
+    def parse(cls, version):
+        if not version[0] in ('v', 'V'):
+            raise ValueError(f"{version}: not a valid semantic version tag. Must start with 'v' or 'V'")
+        self = super(SemVerWithVPrefix, cls).parse(version[1:])
+        return self
+
+    def __str__(self):
+        return 'v' + super(SemVerWithVPrefix, self).__str__()
+
+
+def get_header_defines(header_contents: str) -> Dict[str, str]:
+    header = CppHeaderParser.CppHeader(header_contents, argType='string')
+    defines_dict = {}
+    for d in header.defines_detail:
+        define_tup = d['value'].split(' ', maxsplit=1)
+        defines_dict[define_tup[0]] = define_tup[1]
+    return defines_dict
+
+
+def find_child_text(node) -> Optional[str]:
+    if 'children' in node.keys():
+        if isinstance(node['children'], list):
+            return find_child_text(node['children'][0])
+    elif 'text' in node.keys():
+        return node['text']
+    return None
+
+
+def get_header_version(version_header_contents: str) -> semver.VersionInfo:
+    version_defines = get_header_defines(version_header_contents)
+    try:
+        version_str = next(v.strip('"') for k, v in version_defines.items() if 'version' in k.lower())
+    except StopIteration:
+        die(f'Could not find version define in\n{version_header_contents}: {version_defines}')
+    try:
+        version = SemVerWithVPrefix.parse(version_str)
```

Comment:
put this into the first try. the flow will be terminated with `die` but if this exit function changes some day, this code here will break because version_str is not initialized yet

Reply:
Mmm I don't agree. I want the try/except to explicitly catch exactly the type of error that I'm expecting. If I change it to something like:
```python

    try:
        version_str = next(v.strip('"') for k, v in version_defines.items() if 'version' in k.lower())
        version = SemVerWithVPrefix.parse(version_str)
    except StopIteration:
        die(f'Could not find version define in\n{version_header_contents}: {version_defines}')
    except ValueError as e:
        die(f'SemVer error: {e}')
```
I don't want the first line to raise a `ValueError`, which was intended to catch a semver error.